### PR TITLE
Add travis-ci support on Linux

### LINKED ISSUE #2: no CI workflow
Tracking issue for CI workflow for linux and windows OSes.

### PR DESCRIPTION
**Pull request by minggangw**
Fix #2

### CODE REVIEW

**Comment by minggangw** [comment]
@wayneparrott would you please take a look at the failures when running the tests, thanks!

**Comment by wayneparrott** [comment]
@minggangw thx for setting up travis ci. 

This PR is blocked on issue #8.

tl;dr

Looking at the logs I can see the root issue is the post-install script depends upon `colcon`.  
```
npm WARN lifecycle rclnodejs-cli@0.0.1~postinstall: cannot run in wd rclnodejs-cli@0.0.1 colcon build --base-paths package-creation-tool (wd=/root/rclnodejs-cli)
```

This dependency results from the package's design which is a basic ros2 package, i.e., see package.xml, that includes ros2cli pkg extension implemented in python3. The package-creation tool script runs a node child process of the `ros2cli` extension, e.g., `ros2 pkg create-nodejs`. Back to the colcon dependency, the post-install script run `colcon build` on the rclnodejs-cli package to create the package-creation extension. 

I'm thinking the design should work with colcon being optional.

**Comment by minggangw** [comment]
The `colcon` has been installed through `python3-colcon-common-extensions`, it seems that it should be available?

**Comment by wayneparrott** [comment]
> The colcon has been installed through python3-colcon-common-extensions, it seems that it should be available?

@minggangw  I didn't realize our env includes colcon. I'll continue with the current implementation and get the tests running with the colcon dependency.

**Comment by minggangw** [comment]
`rclnodejs` also uses `colcon` to build a message package, please see https://github.com/RobotWebTools/rclnodejs/blob/ba3b0191471cd3fc3e87de7ef8cffa99f803b1c7/scripts/compile_tests.js#L78, so I think it's feasible to use it here.

**Comment by wayneparrott** [comment]
OK. So returning to the `colcon` dependency.

From the build log the post-install script had an issue running `colcon` which if it ran correctly would create the `rclnodejs-cli/install` folder.  
```
npm WARN lifecycle rclnodejs-cli@0.0.1~postinstall: cannot run in wd rclnodejs-cli@0.0.1 colcon build --base-paths package-creation-tool (wd=/root/rclnodejs-cli)
```
Searching on the error msg I found several articles and SA threads that point to user permission issues when running npm scripts. 
https://medium.com/@manusajith/how-to-fix-npm-install-failed-with-cannot-run-in-wd-7e1b407708ce

**Comment by minggangw** [comment]
I changed to use `npm install --unsafe-perm` in order to run as root, and we have two failures, which looks better :) You can merge this PR in advance if you want to debug based on the current CI configuration.

**Comment by wayneparrott** [comment]
@minggangw  thank you. I have a small fix that I believe will resolve the 2 failing test cases. I made an assumption for the cwd of the test case which is probably incorrect. I will proceed with the merge and will fix the test cases.
